src/v/wasm/wasmedge.cc

```diff
@@ -0,0 +1,692 @@
+                  throw wasm_exception(
+                    ss::format(
+                      "transform execution {} failed", _user_module_name),
+                    errc::user_code_failure);
+              }
+              auto user_result = WasmEdge_ValueGetI32(returns[0]);
+              if (user_result != 0) {
+                  probe->transform_error();
+                  throw wasm_exception(
+                    ss::format(
+                      "transform execution {} resulted in error {}",
+                      _user_module_name,
+                      user_result),
+                    errc::user_code_failure);
+              }
+          });
+    }
+
+    ssx::threaded_work_queue _queue;
+
+    std::vector<WasmEdgeModule> _modules;
+    WasmEdgeStore _store_ctx;
+    WasmEdgeVM _vm_ctx;
+
+    ss::sstring _user_module_name;
+    std::unique_ptr<transform_module> _rp_module;
+    std::unique_ptr<schema_registry_module> _sr_module;
+    std::unique_ptr<wasi::preview1_module> _wasi_module;
+};
+
+WasmEdgeModule create_module(std::string_view name) {
+    auto wrapped = WasmEdge_StringWrap(name.data(), name.size());
+    return WasmEdgeModule(WasmEdge_ModuleInstanceCreate(wrapped));
+}
+
+class wasmedge_engine_factory : public factory {
+public:
+    wasmedge_engine_factory(
+      WasmEdge_ConfigureContext* config_ctx,
+      ssx::thread_worker* t,
+      model::transform_metadata meta,
+      iobuf wasm_module,
+      schema_registry* sr,
+      ss::logger* l)
+      : _config_ctx(config_ctx)
+      , _worker(t)
+      , _sr(sr)
+      , _wasm_module(std::move(wasm_module))
+      , _meta(std::move(meta))
+      , _logger(l) {}
+
+    ss::future<std::unique_ptr<engine>> make_engine() final {
+        WasmEdge_Result result;
+
+        auto store_ctx = WasmEdgeStore(WasmEdge_StoreCreate());
+
+        auto vm_ctx = WasmEdgeVM(
+          WasmEdge_VMCreate(_config_ctx, store_ctx.get()));
+
+        auto wasmedge_transform_module = create_module(transform_module::name);
+        auto xform_module = std::make_unique<transform_module>();
+        register_transform_module(
+          xform_module.get(), wasmedge_transform_module);
+
+        auto wasmedge_sr_module = create_module(schema_registry_module::name);
+        auto sr_module = std::make_unique<schema_registry_module>(_sr);
+        register_sr_module(sr_module.get(), wasmedge_transform_module);
+
+        auto wasmedge_wasi_module = create_module(wasi::preview1_module::name);
+        std::vector<ss::sstring> args{_meta.name()};
+        absl::flat_hash_map<ss::sstring, ss::sstring> env = _meta.environment;
+        env.emplace("REDPANDA_INPUT_TOPIC", _meta.input_topic.tp());
+        env.emplace("REDPANDA_OUTPUT_TOPIC", _meta.output_topics.begin()->tp());
+        auto wasi_module = std::make_unique<wasi::preview1_module>(
+          args, env, _logger);
+        register_wasi_module(wasi_module.get(), wasmedge_wasi_module);
+
+        (void)_worker;
```

Comment:
this line seems like not needed ?

Reply:
thanks removed, we'll add compilation on an alien thread to later.